lib/fizzy/execute.cpp

```diff
@@ -454,10 +454,25 @@ inline bool invoke_function(const FuncType& func_type, const F& func, Instance&
 inline bool invoke_function(const FuncType& func_type, uint32_t func_idx, Instance& instance,
     OperandStack& stack, int depth) noexcept
 {
-    const auto func = [func_idx](Instance& _instance, span<const Value> args, int _depth) noexcept {
-        return execute(_instance, func_idx, args.data(), _depth);
-    };
-    return invoke_function(func_type, func, instance, stack, depth);
+    const auto num_args = func_type.inputs.size();
+    assert(stack.size() >= num_args);
+    span<const Value> call_args{stack.rend() - num_args, num_args};
+
+    const auto ret = execute(instance, func_idx, call_args.begin(), depth + 1);
```

Comment:
Actually, why do we have both `.data` and `begin` when they point to the same thing?

Reply:
Not in `std::span`.